### ISSUE TITLE
Dockerfile language server hangs

### ISSUE THREAD

**Issue by willfrey** [microsoft/vscode-docker#3347]
I'm having the Dockerfile language server hang repeatedly. The only fix is for me to manually kill the process in a terminal.

Does this occur consistently?  Yes
Repro steps:

1. Start writing an instruction in a Dockerfile that requires two or more arguments, e.g.
```Dockerfile
# ...
COPY . .
# ...
```
2. Watch the Dockerfile language server eat up 99% of the CPU

Version: 1.18.0
OS: darwin
OS Release: 21.1.0
Product: Visual Studio Code
Product Version: 1.62.3
Language: en

**Comment by rcjsuen** [comment]
@willfrey Thanks for the bug report. Something similar was reported a few weeks back in #3312 but the other person could not reproduce it anymore. Given that we're not seeing widespread reports of this and a Dockerfile with an instruction that has two arguments are a dime a dozen (like your example, a lot of people use `COPY` to move source code and build artifacts around), I'm not sure it is purely contingent on just an instruction that has two arguments.

Would you be able to share the following information?

1. Please attach a screenshot that shows the 99% CPU usage. Since you are on macOS I presume you were observing this via Activity Monitor?
2. Please share a complete copy of your Dockerfile so I can try to open it and reproduce the problem.

Thank you for your assistance.

**Comment by willfrey** [comment]
@rcjsuen Thanks for the reply!

It seems to be better behaved today but I will keep an eye out for if it acts up again. If it does, I'll follow up on here.

**Comment by bwateratmsft** [comment]
I guess I'll close this for now, please let us know if it reoccurs @willfrey! Thanks!

**Comment by willfrey** [comment]
It's happening 🙃 

```Dockerfile
# syntax=docker/dockerfile:1.3-labs

ARG PYTHON_VERSION=3.9.9
ARG POETRY_VERSION=1.1.12

FROM python:${PYTHON_VERSION} AS python

FROM python AS poetry
SHELL [ "/bin/bash", "-o", "pipefail", "-c" ]
ARG POETRY_VERSION
RUN <<EOF
curl --silent --show-error --location https://install.python-poetry.org | PYTHONUSERBASE=/usr/local XDG_DATA_HOME=/usr/local/libexec python3 -
rm --recursive /root/.cache
EOF
CMD [ "/bin/sh" ]

WORKDIR /
FROM poetry AS requirements
WORKDIR /var/tmp
COPY pyproject.toml poetry.lock ./
RUN poetry export --output /requirements.txt --without-hashes
WORKDIR /

FROM poetry AS build
WORKDIR /var/tmp
COPY pyproject.toml poetry.lock README.md ./
COPY sr csrc
RUN <<EOF
python3 -m venv .venv && poetry build --format wheel
mv dist/*.whl /
rm --recursive .venv dist
EOF
WORKDIR /
```
![Screen Shot 2021-12-07 at 5 50 27 PM](https://user-images.githubusercontent.com/13784361/145118056-173a1a26-cb45-4e28-86ac-0a991b476bd9.png)
![Screen Shot 2021-12-07 at 5 49 27 PM](https://user-images.githubusercontent.com/13784361/145118075-6ddd816e-dc7f-46de-ba54-485653dcf883.png)
![Screen Shot 2021-12-07 at 5 47 15 PM](https://user-images.githubusercontent.com/13784361/145118100-c98de415-a21e-40dd-bc6f-e9a4155e2acd.png)

**Comment by rcjsuen** [comment]
@willfrey Thanks for the screenshots and Dockerfile! WIll investigate...

**Comment by rcjsuen** [comment]
@willfrey I've opened https://github.com/rcjsuen/dockerfile-language-service/issues/99 to track this. The issue seems to be contingent on your build stage being the same name as an image.

Instead of...
```Dockerfile
FROM python:${PYTHON_VERSION} AS python
```
...if you change it to something like...
```Dockerfile
FROM python:${PYTHON_VERSION} AS python-base
```
....or anything else, it should be fine.

Thank you so much for reporting this. Have a great day! :)

**Comment by willfrey** [comment]
Interesting! Thank you for the workaround 🙂

**Comment by bwateratmsft** [comment]
This is now released in Docker extension version 1.19.0.
